DESCRIPTION

```diff
@@ -71,6 +71,7 @@ Imports:
     dplyr,
     IRanges,
     MASS,
+    Matrix,
```

Comment:
Can we avoid this dependency? You could use `DelayedArray` and `crossprod()`

Reply:
So I looked a bit into this. Basically Matrix provides an optimised method for crossprod and sparse array for large adjacency matrices. This functionality is absent in DelayedArray, which focuses on delaying (instead of optimising) matrix operations.

In a nutshell, we have two options:
1. downgrade to base matrix and crossprod if we prioritise reducing package dependencies
2. use Matrix crossprod if we prioritise operations with optimised efficiency

Reply:
For a sparse Matrix of 1000 features by 1000 modules with 80% zeros, the time is about 0.1 s, or 1/10 compared to using base matrix and crossprod. The memory usage is 4 Mb instead of 12 Mb. Efficiency further improves for higher percentages of zeros, whereas it stays the same for base matrix.

Reference: https://alfurka.github.io/2022-08-12-faster-matrix-computation-with-r-using-sparsematrix/

Reply:
Thanks! I think my code resembles your solution.

For some reason, matrix is faster than Sparse Matrix:

```
> time_for_matrix
   user  system elapsed 
  4.194   0.393   4.587 
> time_for_sparse_matrix
   user  system elapsed 
  7.253   1.396   8.744
```

It might be that the object checks take longer for sparse matrix, but I don't really know. Anyway, now it's possible to use both of them (default to basic matrix).